marshmallow/schema.py

```diff
@@ -525,6 +525,16 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
             error_handler = self.handle_error or self.__error_handler__
             if callable(error_handler):
                 error_handler(errors, obj)
+            exc = ValidationError(
+                errors,
+                field_names=self._marshal.error_field_names,
+                fields=self._marshal.error_fields,
+                data=result,
+                **self._marshal.error_kwargs
+            )
+            self.handle_error(exc, result)
```

Comment:
> So, by calling handle_error you're already violating strictness setting. I guess, you do not need to call handle_error at all if not strict.

I have to disagree, documentation says:
 > By default, Schema.dump() and Schema.load() will return validation errors as a dictionary (unless strict mode is enabled).

and
 > You can specify a custom error-handling function for a Schema by overriding the handle_error method.

See [Handling errors](http://marshmallow.readthedocs.io/en/latest/extending.html#handling-errors).

So, regardless of strictness setting, we will have errors. If we have errors, they should be passed to the error handler.

I think that, it is the responsibility of error handler to handle errors correctly. It might comply with strictness setting or not.

 > Here error handler is called second time

Of course, this is a mistake. After fixing this, [handling errors is similar to load method](https://github.com/marshmallow-code/marshmallow/blob/dev/marshmallow/schema.py#L671-L684).